src/server/events/gm/MapleOxQuiz/QuestionStep.java

```diff
@@ -0,0 +1,180 @@
+package server.events.gm.MapleOxQuiz;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+import java.io.IOException;
+
+import okhttp3.*;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+
+import constants.ExpTable;
+import client.MapleCharacter;
+import net.server.Server;
+import server.maps.MapleMap;
+import server.events.gm.core.EventStep;
+import tools.MaplePacketCreator;
+import tools.data.output.MaplePacketLittleEndianWriter;
+import net.SendOpcode;
+
+public class QuestionStep extends EventStep {
+	public static final int MAP_EFFECT = 5120017;
+	public static final int QUESTION_TIMER = 10;
+	public static final String OPEN_TRIVIA_URL = "https://opentdb.com/api.php?amount=20&type=boolean";
+
+	public static final String ERROR_MSG = "Sorry everyone, looks like something went wrong with gathering trivia...";
+	public static final String ROUND_MSG = "Ready?";
+	public static final String CORRECT_MSG = "You got it!";
+	public static final String INCORRECT_MSG = "Better luck next time!";
+
+	private static final int COORD_LEFT_BOUND = -286;
+    private static final int COORD_RIGHT_BOUND = -156;
+    private static final int COORD_LOWER_BOUND = -26;
+    private static final int EXP_PER_CORRECT_ANS = 200;
+
+	MapleMap map;
+	Map<String, Boolean> questions = new HashMap<String, Boolean>();
+	Map<MapleCharacter, Integer> results = new HashMap<MapleCharacter, Integer>();
+
+	public QuestionStep(MapleMap map, int numQuestions) {
+		this.map = map;
+		getProblems(numQuestions);
+	}
+
+	// Impl abstract method
+	protected void executeStep()  throws InterruptedException {
+		// Send error response to players
+		if(questions.size() == 0) {
+			map.startMapEffect(ERROR_MSG, MAP_EFFECT, 7 * 1000);
+			Thread.sleep(8 * 1000);
+			return;
+		}
+
+		// Send questions
+		for(Map.Entry<String, Boolean> entry : questions.entrySet()) {
+			map.startMapEffect(ROUND_MSG, MAP_EFFECT, 2 * 1000);
+			Thread.sleep(3 * 1000);
+			sendQuestion(entry.getKey());
+			Thread.sleep(QUESTION_TIMER * 1000);
+			removeClock();
+			checkAnswer(entry.getValue());
+			Thread.sleep(8 * 1000);
+		}
+
+		// Show Ranking
+		showRanking();
+	}
+
+	private void getProblems(int numQuestions) {
```

Comment:
numQuestions is unused, how did you keep track of the number of questions asked?

Reply:
good catch, using string.format passed in num questions